internal/updater/updater_test.go

```diff
@@ -0,0 +1,256 @@
+	})
+})
+
+var _ = Describe("EnsureCondition", func() {
+	var status *rukpakv1alpha1.BundleStatus
+	var condition, anotherCondition metav1.Condition
+
+	BeforeEach(func() {
+		status = &rukpakv1alpha1.BundleStatus{}
+		condition = metav1.Condition{Type: "Working"}
+		anotherCondition = metav1.Condition{Type: "Completed"}
+	})
+
+	It("should add Condition if not present", func() {
+		Expect(updater.EnsureCondition(condition)(status)).To(BeTrue())
+		status.Conditions[0].LastTransitionTime = metav1.Time{}
+		Expect(status.Conditions[0]).To(Equal(condition))
+	})
+
+	It("should return false for no update", func() {
+		status = &rukpakv1alpha1.BundleStatus{Conditions: []metav1.Condition{condition}}
+		Expect(updater.EnsureCondition(condition)(status)).To(BeFalse())
+		Expect(status.Conditions[0]).To(Equal(condition))
+	})
+
+	It("should add Condition if same type not present", func() {
+		status = &rukpakv1alpha1.BundleStatus{Conditions: []metav1.Condition{condition}}
+		Expect(updater.EnsureCondition(anotherCondition)(status)).To(BeTrue())
+		status.Conditions[1].LastTransitionTime = metav1.Time{}
+		Expect(status.Conditions[1]).To(Equal(anotherCondition))
+	})
+})
+
+var _ = Describe("EnsureObservedGeneration", func() {
+	var status *rukpakv1alpha1.BundleStatus
+
+	BeforeEach(func() {
+		status = &rukpakv1alpha1.BundleStatus{}
+	})
+
+	It("should add ObservedGeneration if not present", func() {
+		Expect(updater.EnsureObservedGeneration(3)(status)).To(BeTrue())
+		Expect(status.ObservedGeneration).To(Equal(int64(3)))
+	})
+
+	It("should return false for no update", func() {
+		status.ObservedGeneration = 5
+		Expect(updater.EnsureObservedGeneration(5)(status)).To(BeFalse())
+		Expect(status.ObservedGeneration).To(Equal(int64(5)))
+	})
+})
+
+var _ = Describe("SetPhase", func() {
+	var status *rukpakv1alpha1.BundleStatus
+
+	BeforeEach(func() {
+		status = &rukpakv1alpha1.BundleStatus{}
+	})
+
+	It("should set phase if not present", func() {
+		Expect(updater.SetPhase("phase")(status)).To(BeTrue())
+		Expect(status.Phase).To(Equal("phase"))
+	})
+
+	It("should return false for no update", func() {
+		status.Phase = "phase"
+		Expect(updater.SetPhase("phase")(status)).To(BeFalse())
+		Expect(status.Phase).To(Equal("phase"))
+	})
+})
+
+var _ = Describe("SetBundleInfo", func() {
+	var status *rukpakv1alpha1.BundleStatus
+	var info *rukpakv1alpha1.BundleInfo
+
+	BeforeEach(func() {
+		status = &rukpakv1alpha1.BundleStatus{}
+		info = &rukpakv1alpha1.BundleInfo{}
+	})
+
+	It("should set phase if not present", func() {
+		Expect(updater.SetBundleInfo(info)(status)).To(BeTrue())
+		Expect(status.Info).To(Equal(info))
+	})
+
+	It("should return false for no update", func() {
+		status.Info = info
+		Expect(updater.SetBundleInfo(info)(status)).To(BeFalse())
+		Expect(status.Info).To(Equal(info))
+	})
+})
+
+var _ = Describe("UnsetBundleInfo", func() {
+	var status *rukpakv1alpha1.BundleStatus
+	var info *rukpakv1alpha1.BundleInfo
```

Comment:
nit: would it improve readability if the variable was named something like `emptyInfo` or `nilInfo`?